src/test/perf/contention/contention.cc

```diff
@@ -86,7 +86,7 @@ void test_tasks_f(size_t id)
 
     *res = size;
     size_t* out =
-      contention[n % swapsize].exchange(res, std::memory_order_relaxed);
+      contention[n % swapsize].exchange(res, std::memory_order_release);
```

Comment:
I think you are right, but shouldn't it be `std::memory_order_acq_rel` as this performs as a read as well, that needs to induce the happens-before edge?

Reply:
Yea, I think you're right.  I think I thought the data dependencies in this thread would be sufficient, but they're not.